### ISSUE TITLE
Migrate away from MangoPay?

### ISSUE THREAD

**Issue by Changaco** [liberapay/liberapay.com#374]
At some point during the recent disagreement with MangoPay (which resulted in https://github.com/liberapay/salon/issues/52) I wondered if they had competitors we could potentially switch to. A simple web search led me to the blog post [Quel prestataire de paiement pour compte de tiers choisir ? Comparatif HiPay, LemonWay, Limonetik, MangoPay, Ingenico, PayPal & Braintree](http://www.pequi.eu/quel-prestataire-de-paiement-pour-compte-de-tiers-choisir/) (here's [Google's English Translation](https://translate.google.fr/translate?sl=fr&tl=en&js=y&prev=_t&hl=fr&ie=UTF-8&u=http%3A%2F%2Fwww.pequi.eu%2Fquel-prestataire-de-paiement-pour-compte-de-tiers-choisir%2F&edit-text=&act=url) if you don't speak French).

TL;DR the first two look interesting: [HiPay](http://www.hipayfullservice.fr/) and [LemonWay](https://www.lemonway.fr/).

***Update:** [Stripe Connect](https://stripe.com/connect) is also becoming interesting.*

Moving away from MangoPay would be a lot of work, so the competitor would have to be significantly better. For example:
- lower fees
- a policy more in line with our own neutrality
  - I've already glanced at LemonWay's contract, they're no better than MangoPay on this
- better support of currencies (for #182)
  - support for currency conversion
  - simpler fees than MangoPay

Contracting with another payment operator would involve the legal entity, so ping @liberapay/codirectors.

**Comment by MartinDelille** [comment]
I would go for the service that would offer support/contribution to Liberapay. Are they aware of the project?

**Comment by SimonSarazin** [comment]
Helloasso use payzen (i should check if it's still the case ) and Ismaël
said me that for big amounts the fee is really low.

Le sam. 9 juil. 2016 11:37, Martin Delille <notifications@github.com> a
écrit :

> I would go for the service that would offer support/contribution to
> Liberapay. Are they aware of the project?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/liberapay/liberapay.com/issues/374#issuecomment-231525742>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAl2ju6GUJBmZTt88mP4sD_B6YGKZ8ZUks5qT2vOgaJpZM4JIk3r>
> .
>
-- 

Simon
0684284240
http://simons.fr

**Comment by Changaco** [comment]
PayZen is a different kind of payment processor, it's not suitable for platforms like Liberapay.

**Comment by Changaco** [comment]
@MartinDelille I haven't contacted anyone yet, so no, they're probably not aware of Liberapay.

What kind of support/contribution did you have in mind?

**Comment by MartinDelille** [comment]
Liberapay is a nice tool so people from these company could get more involved in the project for personnal/ethical interest.

**Comment by Changaco** [comment]
@SimonSarazin forwarded me an email concerning payment processors in which someone he knows said we should look into LemonWay if we have a low volume (<2M€/year, we're definitely in that category) and HiPay otherwise. The message also strongly recommended negotiating for lower fees.

**Comment by aggsol** [comment]
Having multiple payment processors would also add redundancy and mitigate pressure from having just one. As a first step adding more instead of replacing Mangopay would be IMO the way to go.

**Comment by Changaco** [comment]
Handling multiple payment processors in parallel would be more work than replacing MangoPay with another one. The additional complexity would be similar and possibly higher than the management of multiple currencies.

**Comment by Changaco** [comment]
It looks like [Braintree Marketplace](https://www.braintreepayments.com/products/braintree-marketplace) is no longer restricted to platforms based in the US. However I haven't seen anything on their website about compliance with EU licensing requirements for marketplaces, so at this point I'm not sure we can legally use them without getting our own license.

**Comment by Changaco** [comment]
If we were to integrate PayPal then their [Adaptive Payments](https://developer.paypal.com/docs/classic/products/adaptive-payments/) seem to be what we would need. There are no licensing issues with this as far as I can tell, because we would be transferring donations between PayPal accounts (so PayPal holds the money, not us).

**Comment by rpbaptist** [comment]
I would not recommend Lemonway. In fact, we are switching away from them.

Their API is not restful at all. You will always get http 200 back and have to dig in the JSON/XML response with single character key values to find out what actually happened.

What's worse is that each currency requires a completely separate environment which also means you will have to do everything twice and mirror their data to keep everything in sync.

Their web interface is incredibly slow and unreliable. It has only client side validation for some forms which is a security risk. I reported this to them and they never did anything with it.

On top of that they are not service orientated at all. Very bureacratic and "computer says no" attitude.

There is a high turnover in personal there as well which is also a red flag. I would definitely not recommend them.

**Comment by Changaco** [comment]
I have just received an email from mangopay saying that they want to terminate their contract with us because we have/had a few controversial users.

**Comment by Changaco** [comment]
I've replied, asking for confirmation, and information on how much time we have left.

**Comment by Changaco** [comment]
I've discovered a payment solution that seems more advanced than most of the others I've seen so far: [Treezor - Bank as a Service Platform](https://www.treezor.com/en/). A Liberapay account is like a "bank account for donations", so "Banking as a Service" may fit our needs well.

**Comment by Changaco** [comment]
[This blog post in French](https://startup.info/fr/treezor/) has some extra info from Treezor. As I expected the section about pricing mentions possible set-up and monthly costs in addition to transaction fees. The section about competitors lists these:

- https://www.wirecard.com/
- http://www.bnkbl.com/
- https://www.solarisbank.com/en/
- https://prepaidfinancialservices.com/en/
- https://www.contis.com/

**Comment by Changaco** [comment]
Another e-wallet provider (same category as HiPay): [Adyen MarketPay](https://www.adyen.com/our-solution/online-payments/marketplaces).

**Comment by albjeremias** [comment]
Why you didn't had fair-coin as a possibility? https://fair-coin.org/en/use-cases 
checkout fairpay:
http://fairpay.fair.coop/

**Comment by revi** [comment]
I don't think we are migrating away to cryptocurrency...

What we need here is a [Payment Processor](https://en.wikipedia.org/wiki/Payment_processor), not a blockchain solution.

**Comment by tuxayo** [comment]
@albjeremias It would be a nice addition, but IMHO support for EUR and USD at least it absolutely necessary. Otherwise step would be too huge for users.
It's already hard to motivate most libre/open source software supporters to setup recurring donations.

**Comment by CameronNemo** [comment]
@revi there are cryptocurrency payment processors that do automatic settlement in fiat currency. BitPay can do Euros and US dollars, and they have a 1% processing fee. Obviously this would not be a direct replacement for MangoPay, but it could be a nice alternative to support.

This reddit thread shows people struggling to use mangopay: https://www.reddit.com/r/linux/comments/8wvuhh/liberapay_a_nonprofit_and_opensource_recurrent/e1yu8w9/

Something tells me these people wouldn't find paying with crypto too onerous.

**Comment by mattbk** [comment]
https://liberapay.com/about/stats shows that 1710 people figured out how to use Liberapay without having issues with Mangopay as of last week. I've paid in to Liberapay from the US and didn't have any issues--but I have no money in cryptocurrency and that would mean an extra step for me.

**Comment by MRZA-MRZA** [comment]
BitPay was mentioned above (they support BCH!). IMO nowadays it's absolutely necessary to support payments in cryptocurrency.

**Comment by Changaco** [comment]
I'm impressed by Adyen's documentation, they seem to have built a great payment processing system.

**Comment by droidmonkey** [comment]
Stripe Connect seems like the most prolific and mainstream out of the bunch.

**Comment by fraang** [comment]
What kind of controversial users are we talking about here?

**Comment by Changaco** [comment]
Got an email from someone at [Railsbank](https://www.railsbank.com/) who saw the Hacker News thread.

**Comment by Changaco** [comment]
Adyen says Liberapay falls within their [Restricted & Prohibited List](https://docs.adyen.com/legal/adyen-restricted-prohibited-list), so they won't help us.

**Comment by MartinDelille** [comment]
How is it going with [Railsbank](https://www.railsbank.com/)?

**Comment by Changaco** [comment]
Partially relevant article in French: [Comment bien choisir sa solution de paiement](https://www.journaldunet.com/economie/finance/1210743-comment-bien-choisir-sa-solution-de-paiement/).

**Comment by mattbk** [comment]
> This blog post in French has some extra info from Treezor. As I expected the section about pricing mentions possible set-up and monthly costs in addition to transaction fees. 

@Changaco, did you contact Treezor directly to ask about fees and subscription costs? If they provide a good replacement e-wallet provider, paying reasonable additional costs seem warranted to keep Liberapay afloat and growing.

**Comment by Changaco** [comment]
@mattbk I sent them a message in July, never got a response. Wallets are a dead end, so it doesn't matter.

**Comment by mattbk** [comment]
> Wallets are a dead end, so it doesn't matter.

I assume you're referring to your comments at https://github.com/liberapay/liberapay.com/issues/1171#issuecomment-426244561?

**Comment by Changaco** [comment]
@mattbk Not really, that comment doesn't say that wallets are a dead end, it only says that we don't really need them. The reason why wallets are a dead end is that they're too similar to bank accounts, and Liberapay can't operate like a bank because it's just a small nonprofit with zero employees. Also there's no such thing as a bank that operates worldwide, but Liberapay needs to be available in as many countries as possible because our users are all over the world. (Relevant toot: <https://mastodon.xyz/@Liberapay/100883660682359307>.)

**Comment by mattbk** [comment]
Thanks for the explanation. I appreciate it.

**Comment by puneetkumartripathi** [comment]
Thanks for helping. [startup stories](https://www.startupurban.com/)

**Comment by corentinclichy** [comment]
Hello,

I came across this thread. I'm currently looking into PSPs in France that support both C2C and B2B. In the end, which solution did you choose?
